### ISSUE TITLE
Specify the scope of Cosmos DB connection strings in Terraform output

### ISSUE THREAD

**Issue by Sarah-Aly** [terraform-providers/terraform-provider-azurerm#9621]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Scope of Cosmos DB connection strings needs to be specified in Terraform output to reflect the same level of details that Azure CLI provides.
Currently Terraform spits an array of connection strings that looks like below:
```json
{
        "name": "CosmosDB_Connection_String",
        "slotSetting": false,
        "value": [
            "AccountEndpoint=https://<redacted>.documents.azure.com:443/;AccountKey=<redacted>",
            "AccountEndpoint=https://<redacted>.documents.azure.com:443/;AccountKey=<redacted>",
            "AccountEndpoint=https://<redacted>.documents.azure.com:443/;AccountKey=<redacted>",
            "AccountEndpoint=https://<redacted>.documents.azure.com:443/;AccountKey=<redacted>"
        ]
    }
```
Output example when using `az cosmosdb keys list --type connection-strings`
```json
{
  "connectionStrings": [
    {
      "connectionString": "AccountEndpoint=https://<redacted>.documents.azure.com:443/;AccountKey=<redacted>",
      "description": "Primary SQL Connection String"
    },
    {
      "connectionString": "AccountEndpoint=https://<redacted>.documents.azure.com:443/;AccountKey=<redacted>",
      "description": "Secondary SQL Connection String"
    },
    {
      "connectionString": "AccountEndpoint=https://<redacted>.documents.azure.com:443/;AccountKey=<redacted>",
      "description": "Primary Read-Only SQL Connection String"
    },
    {
      "connectionString": "AccountEndpoint=https://<redacted>.documents.azure.com:443/;AccountKey=<redacted>",
      "description": "Secondary Read-Only SQL Connection String"
    }
  ]
}
```

### New or Affected Resource(s)

* azurerm_cosmosdb_account

### References

* https://docs.microsoft.com/en-us/cli/azure/cosmosdb?view=azure-cli-latest#az_cosmosdb_list_connection_strings

**Comment by tombuildsstuff** [comment]
Taking a look through here it appears the `connection_strings` block should be deprecated and replaced by two top-level properties to match the others: e.g.:

* `primary_connection_string`
* `secondary_connection_string`
* `primary_readonly_connection_string`
* `secondary_readonly_connection_string`

**Comment by favoretti** [comment]
@tombuildsstuff how do you wanna break them up, based on description? Wouldn't that be a bit too flaky?

**Comment by tombuildsstuff** [comment]
@favoretti I don't see another option unfortunately, we'll have to do this conditionally like so:

```
primaryReadOnlyConnectionString := ""
if v, ok := connectionStrings["Primary Read-Only SQL Connection String"]; ok {
  primaryReadOnlyConnectionString = v.(string)
}
```

which should avoid any breaking changes to this on the Azure side, by setting an empty string by default - WDYT?

**Comment by favoretti** [comment]
Yeah, that's the only way I saw as well, unfortunately, but now I'm kinda wondering how many combinations would there be, since that one has "SQL" in it. Mongo would have its own set, gremlin its own, etc, etc?

**Comment by tombuildsstuff** [comment]
@favoretti sorry missed this ping - agreed it's possible there's multiple combinations available here, but I think we could likely add those as folks need them, others could make sense on the other resources too (e.g. on `cosmosdb_table` as in #9990), WDYT?

**Comment by VRabadan** [comment]
Just came across this. I think that at least `primary_connection_string` would be useful to use in other parts of the terraform file. Current list of values is confusing to use.
